fw/http.c

```diff
@@ -4847,9 +4867,17 @@ tfw_h1_resp_adjust_fwd(TfwHttpResp *resp)
 	 * cache it can be dropped.
 	 */
 	if (unlikely(test_bit(TFW_HTTP_B_REQ_DROP, req->flags))) {
+		TfwCliConn *cli_conn = (TfwCliConn *)req->conn;
+
 		T_DBG2("%s: resp=[%p] dropped: client disconnected\n",
 		       __func__, resp);
 		tfw_http_resp_pair_free(req);
+		/*
+		 * Put the websocket server connection when client connection is
+		 * lost after successful upgrade request.
+		 */
+		if (unlikely(cli_conn->proto.type & TFW_FSM_WEBSOCKET))
+			tfw_connection_put(cli_conn->pair);
```

Comment:
I'd move the conditional put() to a new function, just to make the code more readable

Reply:
Done